base/dict.jl

```diff
@@ -75,12 +75,30 @@ function showdict{K,V}(io::IO, t::Associative{K,V}; compact = false)
                 print(io, typeof(t))
             end
             print(io, '(')
+
+            # @Duplicate of code in show(IO, Pair)
+            #
+            # Previously called `show` to print each Pair for the brackets
+            # that it inserted if either element of the Pair was itself a Pair.
+            # However, it calls `show_default` if the actual types of either element
+            # does not match the type parameters of the Pair.
+            # e.g: Pair{Any,Any}(1, 2)
+            print_pair_with_brackets = (io::IO, p::Pair) -> begin
```

Comment:
@vtjnash Hmm... That is a fair bit easier to read - would it still qualify as a compact form though?... :)
It also might be better to put that into another PR? @Keno